company-box-doc.el

```diff
@@ -118,14 +124,15 @@
       (make-frame-visible (frame-parameter nil 'company-box-doc-frame)))))
 
 (defun company-box-doc (selection frame)
-  (-some-> (frame-parameter frame 'company-box-doc-frame)
-           (make-frame-invisible))
-  (when (timerp company-box-doc--timer)
-    (cancel-timer company-box-doc--timer))
-  (setq company-box-doc--timer
-        (run-with-idle-timer
-         company-box-doc-delay nil
-         (lambda nil (company-box-doc--show selection frame)))))
+  (if company-box-doc-enable
+      (-some-> (frame-parameter frame 'company-box-doc-frame)
+               (make-frame-invisible))
```

Comment:
I don't think that the code reflect your intention. if `company-box-doc-enable` is non-nil, it will hide the frame, not prevent the doc to appear

Reply:
Aha, Oops, I updated it to `when`.